cmd/verify/network/cmd.go

```diff
@@ -0,0 +1,234 @@
+/*
+Copyright (c) 2023 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"os"
+	"strings"
+	"time"
+
+	"github.com/briandowns/spinner"
+	"github.com/spf13/cobra"
+
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	"github.com/openshift/rosa/pkg/arguments"
+	"github.com/openshift/rosa/pkg/aws"
+	"github.com/openshift/rosa/pkg/ocm"
+	"github.com/openshift/rosa/pkg/reporter"
+	"github.com/openshift/rosa/pkg/rosa"
+)
+
+var args struct {
+	region     string
+	roleArn    string
+	statusOnly bool
+	subnetIDs  []string
+	watch      bool
+}
+
+var Cmd = &cobra.Command{
+	Use:  "network",
+	Long: "Verify that the VPC subnets are configured correctly.",
+	Example: `  # Verify two subnets
+  rosa verify network --subnet-ids subnet-03046a9b92b5014fb,subnet-03046a9c92b5014fb`,
+	Run: run,
+}
+
+type NetworkVerifyState string
+
+const (
+	clusterFlag    = "cluster"
+	roleArnFlag    = "role-arn"
+	statusOnlyFlag = "status-only"
+	subnetIDsFlag  = "subnet-ids"
+	watchFlag      = "watch"
+
+	NetworkVerifyPending NetworkVerifyState = "pending"
+	NetworkVerifyPassed  NetworkVerifyState = "passed"
+	NetworkVerifyFailed  NetworkVerifyState = "failed"
+
+	delay = 5 * time.Second
+)
+
+func init() {
+	flags := Cmd.Flags()
+
+	ocm.AddOptionalClusterFlag(Cmd)
+
+	flags.StringSliceVar(
+		&args.subnetIDs,
+		subnetIDsFlag,
+		nil,
+		"The Subnet IDs to verify. "+
+			"Format should be a comma-separated list.",
+	)
+
+	arguments.AddRegionFlag(flags)
+
+	flags.StringVar(
+		&args.roleArn,
+		roleArnFlag,
+		"",
+		"STS Role ARN with get secrets permission.",
+	)
+
+	flags.BoolVarP(
+		&args.watch,
+		watchFlag,
+		"w",
+		false,
+		"Watch network verification progress.",
+	)
+
+	flags.BoolVarP(
+		&args.statusOnly,
+		statusOnlyFlag,
+		"s",
+		false,
+		"Check status of previously submitted subnets.",
+	)
+}
+
+func run(cmd *cobra.Command, _ []string) {
+	r := rosa.NewRuntime().WithAWS().WithOCM()
+	defer r.Cleanup()
+
+	if cmd.Flags().Changed(subnetIDsFlag) {
+		if cmd.Flags().Changed(clusterFlag) {
+			r.Reporter.Errorf("One and only one of %s or %s is required",
+				clusterFlag, subnetIDsFlag)
+			os.Exit(1)
+		}
+
+		region, err := aws.GetRegion(arguments.GetRegion())
+		if err != nil {
+			r.Reporter.Errorf("Error getting region: %v", err)
+			os.Exit(1)
+		}
+		args.region = region
+	} else if cmd.Flags().Changed(clusterFlag) {
+		if cmd.Flags().Changed(subnetIDsFlag) {
```

Comment:
Couldn't this check and the one above be merged into a single condition which checks if both have changed?

Reply:
I don't mind that much, but personally find it cleaner